app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -28,7 +29,15 @@ class Api::V2::SubscriptionsController < Api::V2::ApiController
     param :match_installed, :bool, :desc => N_("Return subscriptions that match installed products of the specified host")
     param :no_overlap, :bool, :desc => N_("Return subscriptions which do not overlap with a currently-attached subscription")
     def index
-      respond(:collection => scoped_search(index_relation.uniq, :cp_id, :asc, :resource_class => Pool, :includes => [:subscription]))
+      collection = scoped_search(
+        index_relation.uniq, :cp_id, :asc, resource_class: Pool, includes: [:subscription])
+      if params[:activation_key_id]
+        key_pools = @activation_key.get_key_pools
+        collection[:results] = collection[:results].map do |pool|
+          pool.with_quantity_attached(key_pools)
```

Comment:
@jlsherrill I had it passing in the `@activation_key`, but I couldn't figure out a good way to cache the key_pools since each Pool object has different instance variables. This implementation calls `get_key_pools` only once.